charts/newrelic-infrastructure-v3/templates/control-plane/configmap.yaml

```
@@ -1,17 +1,16 @@
 {{- if .Values.controlPlane.enabled -}}
+---
 apiVersion: v1
 kind: ConfigMap
 metadata:
-  namespace: {{ .Release.Namespace }}
   labels:
     {{- include "newrelic.labels" . | nindent 4 }}
   name: {{ template "newrelic.fullname" . }}-control-plane
+  namespace: {{ .Release.Namespace }}
 data:
   nri-kubernetes.yml: |
-    {{- .Values.common.config | toYaml | nindent 4 }}
+    {{- include "newrelic.commonIntegrationConfig" . | nindent 4 }}
```

This might look a bit cryptic. WDYT about making the helper return the overrides? So we can do something like:

```suggestion
    {{- merge (.Values.common.config (include "newrelic.configDefaults" . | fromYaml) ) | toYaml | nindent 4 }}
```

---

I am like against of having "logic" in the template and just return from the helper what is has to be in the template. That is the reason I did that. But I have to say that your approach seems cleaner and easier to understand.

Done.